### PR TITLE
Implement e2e test for tealdbg

### PR DESCRIPTION
**Pull request by algorandskiy**
## Summary

Tealdbg test using CDT mock program instead of Google Chrome:
* Run debugger
* Run CDT simulator, connect to the debugger, send control messages
* Wait for termination, check output

## Test Plan

This is a test

### CODE REVIEW

File: cmd/tealdbg/cdtmock/main.go [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+	}
+	copy(buf, msg)
+	return len(msg), nil
+}
+
+func (c *wsClient) Close() {
+	c.conn.WriteControl(websocket.CloseMessage, websocket.FormatCloseMessage(websocket.CloseNormalClosure, ""), time.Now().Add(5*time.Second))
+	c.conn.CloseWithoutFlush()
+}
+
```
Comment (egieseke) [inline cmd/tealdbg/cdtmock/main.go:73]: Consider adding a comment here to explain the purpose.

**Review by egieseke — APPROVED** [review]
Looks good.

File: test/e2e-go/cli/goal/expect/tealdbgTest.exp [hunk 2]
```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/expect -f
+set err 0
+log_user 1
+
+if { [catch {
+    source  goalExpectCommon.exp
+
```
Comment (egieseke) [inline test/e2e-go/cli/goal/expect/tealdbgTest.exp:7]: Suggest increasing timeout from the default of 30 seconds to 60 seconds.

    `set timeout 60`

Comment (algorandskiy) [reply test/e2e-go/cli/goal/expect/tealdbgTest.exp:7]: done

**Review by tsachiherman — COMMENTED** [review]
All looks good, only thing I would change is to move `test/e2e-go/cli/goal/expect/tealdbgTest.exp` -> `test/e2e-go/cli/tealdbg/expect/tealdbgTest.exp`
and the `cmd/tealdbg/cdtmock` to  `test/e2e-go/cli/tealdbg/cdtmock/`

**Comment by algorandskiy** [comment]
Done

File: cmd/tealdbg/cdtmock/main.go (hunk 1 above)
Comment (algorandskiy) [reply cmd/tealdbg/cdtmock/main.go]: moved to test harness per Tsachi suggestion

File: test/e2e-go/cli/tealdbg/expect/tealdbg_expect_test.go [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+package expect
+
+import (
+	"testing"
+
+	"github.com/algorand/go-algorand/test/framework/fixtures"
+)
+
+// TestAlgohWithExpect Process all expect script files with suffix Test.exp within the test/e2e-go/cli/tealdbg/expect directory
+func TestTealdbgWithExpect(t *testing.T) {
```
Comment (tsachiherman) [inline test/e2e-go/cli/tealdbg/expect/tealdbg_expect_test.go:25]: I'm surprised the `go vet` didn't complain that the function name in the description doesn't match the function name in the code.

**Review by tsachiherman — APPROVED** [review]
One small typo in comment; You'll be able to fix that in a follow up PR.

**Pull request merged**